From be290de24214538f806ef68ab716bd0752560ff3 Mon Sep 17 00:00:00 2001
From: "Ch1llb0/skin.osmc" <36623079+Ch1llb0@users.noreply.github.com>
Date: Tue, 12 Mar 2019 13:46:54 +0100
Subject: [PATCH] PVR windows fix

MusicVisualisation.xml:
- fix bigger OSD cover art to prevent overlap with music OSD

MyPVRChannels.xml:
- remove onright of the list
- add scrollbar control to sub-menu

MyPVRGuide.xml:
- add scrollbar
- fix onleft, onright and onback of EPG grid
- add fade effect to labels when not focused
- add second sub-menu indicator
- add scrollbar control to sub-menu

MyPVRRecordings.xml:
- remove onright of the list
- add scrollbar control to sub-menu
- fix watched toggle

MyPVRSearch.xml:
- remove onright of the list
- add scrollbar control to sub-menu

MyPVRTimers.xml:
- remove onright of the list
- add scrollbar control to sub-menu
---
 16x9/MusicVisualisation.xml |  4 +-
 16x9/MyPVRChannels.xml      | 20 ++++++++-
 16x9/MyPVRGuide.xml         | 83 ++++++++++++++++++++++++++++++-------
 16x9/MyPVRRecordings.xml    | 24 +++++++++--
 16x9/MyPVRSearch.xml        | 21 +++++++++-
 16x9/MyPVRTimers.xml        | 21 +++++++++-
 6 files changed, 146 insertions(+), 27 deletions(-)

diff --git a/16x9/MusicVisualisation.xml b/16x9/MusicVisualisation.xml
index 70a126a..0106184 100755
--- a/16x9/MusicVisualisation.xml
+++ b/16x9/MusicVisualisation.xml
@@ -35,8 +35,8 @@
 			<control type="image">
 				<right>980</right>
 				<centertop>50%</centertop>
-				<width>700</width>
-				<height>700</height>
+				<width>580</width>
+				<height>580</height>
 				<fadetime>200</fadetime>
 				<texture background="true">$INFO[MusicPlayer.Cover]</texture>
 				<aspectratio aligny="center">keep</aspectratio>
diff --git a/16x9/MyPVRChannels.xml b/16x9/MyPVRChannels.xml
index 2de497d..cdba475 100755
--- a/16x9/MyPVRChannels.xml
+++ b/16x9/MyPVRChannels.xml
@@ -71,7 +71,6 @@
 				<onup>50</onup>
 				<ondown>50</ondown>
 				<onleft>60</onleft>
-				<onright condition="!Skin.HasSetting(KioskMode)">3001</onright>
 				<pagecontrol>60</pagecontrol>
 				<focusposition>3</focusposition>
 				<orientation>vertical</orientation>
@@ -302,11 +301,28 @@
 					<scrolltime>200</scrolltime>
 					<orientation>vertical</orientation>
 					<itemgap>0</itemgap>
-					<defaultcontrol always="true">99</defaultcontrol>
+					<defaultcontrol always="true">3</defaultcontrol>
 
 					<include content="SideMenuAnimation">
 						<param name="containerID">3000</param>
 					</include>
+					
+					<!-- Scrollbars -->
+					<control type="radiobutton" id="1">
+						<label>31005</label>
+						<height>52</height>
+						<onclick>Skin.ToggleSetting(Scrollbars)</onclick>
+						<selected>!Skin.HasSetting(Scrollbars)</selected>
+					</control>
+
+
+					<control type="image" id="2">
+						<left>0</left>
+						<top>0</top>
+						<width>410</width>
+						<height>10</height>
+						<texture>transparent.png</texture>
+					</control>
 
 					<!-- Sort by -->
 					<control type="button" id="3">
diff --git a/16x9/MyPVRGuide.xml b/16x9/MyPVRGuide.xml
index 5734f61..1b292ee 100755
--- a/16x9/MyPVRGuide.xml
+++ b/16x9/MyPVRGuide.xml
@@ -17,6 +17,25 @@
 			<control type="group">
 				<visible>Control.IsVisible(10)</visible>
 
+				<!-- Scrollbar (list) -->
+				<control type="scrollbar" id="60">
+					<left>70</left>
+					<top>230</top>
+					<width>20</width>
+					<height>552</height>
+					<onleft condition="!Skin.HasSetting(KioskMode)">3000</onleft>
+					<onright>10</onright>
+					<showonepage>false</showonepage>
+					<orientation>vertical</orientation>
+					<colordiffuse>$VAR[OverlayColorNF]</colordiffuse>
+					<texturesliderbackground border="11,1,1,1">common/ScrollBackground.png</texturesliderbackground>
+					<texturesliderbar border="11,1,1,1">common/ScrollbarGripNF.png</texturesliderbar>
+					<texturesliderbarfocus border="11,1,1,1" colordiffuse="$VAR[DialogColor1]">common/ScrollbarGripFO.png</texturesliderbarfocus>
+					<textureslidernib></textureslidernib>
+					<textureslidernibfocus></textureslidernibfocus>
+					<visible>!Skin.HasSetting(Scrollbars)</visible>
+				</control>
+				
 				<!-- EPG Grid -->
 				<control type="epggrid" id="10">
 					<left>120</left>
@@ -25,11 +44,12 @@
 					<height>606</height>
 					<onup>10</onup>
 					<ondown>10</ondown>
-					<onleft>3000</onleft>
-					<onright>3000</onright>
-					<onback>3000</onback>
+					<onleft>10</onleft>
+					<onright>10</onright>
+					<onback>60</onback>
 					<rulerunit>12</rulerunit>
 					<timeblocks>54</timeblocks>
+					<pagecontrol>60</pagecontrol>
 					<viewtype label="19032">list</viewtype>
 					<progresstexture border="4,61,4,4">pvr/ProgressTimeline.png</progresstexture>
 					<scrolltime tween="sine" easing="out">240</scrolltime>
@@ -97,6 +117,7 @@
 							<height>69</height>
 							<label>[B]$INFO[ListItem.ChannelName][/B]</label>
 							<textcolor>$VAR[TextColor1]</textcolor>
+							<animation effect="fade" start="100" end="50" time="200" tween="cubic" easing="out" reversible="true" condition="!Control.HasFocus(10)">Conditional</animation>
 						</control>
 
 						<!-- Channel number -->
@@ -107,6 +128,7 @@
 							<align>right</align>
 							<label>$INFO[ListItem.ChannelNumberLabel]</label>
 							<textcolor>$VAR[TextColor1]</textcolor>
+							<animation effect="fade" start="100" end="50" time="200" tween="cubic" easing="out" reversible="true" condition="!Control.HasFocus(10)">Conditional</animation>
 						</control>
 						
 						<!-- Highlighting -->
@@ -184,6 +206,7 @@
 							<height>69</height>
 							<label>[B]$INFO[ListItem.Label][/B]</label>
 							<textcolor>$VAR[TextColor1]</textcolor>
+							<animation effect="fade" start="100" end="50" time="200" tween="cubic" easing="out" reversible="true" condition="!Control.HasFocus(10)">Conditional</animation>
 						</control>
 
 						<!-- Recording image -->
@@ -341,18 +364,31 @@
 		
 		<!-- Submenu indicator -->
 		<control type="group">
-				<visible>!ControlGroup(9002).HasFocus + !Window.IsVisible(contextmenu) + !Skin.HasSetting(KioskMode)</visible>
-				<animation effect="fade" start="0" end="100" time="200">VisibleChange</animation>
-				<animation effect="slide" time="200" start="-200,0" end="0,0">WindowOpen</animation>
-				<animation effect="slide" time="200" start="0,0" end="-200,0">WindowClose</animation>
-				<control type="image">
-					<left>10</left>
-					<centertop>50%</centertop>
-					<width>30</width>
-					<height>30</height>
-					<texture colordiffuse="$VAR[DialogColor2]">sub-menu-return.png</texture>
-				</control>
+			<visible>!Window.IsVisible(contextmenu) + !Skin.HasSetting(KioskMode) + Control.HasFocus(10)</visible>
+			<animation effect="fade" start="0" end="100" time="200">VisibleChange</animation>
+			<animation effect="slide" time="200" start="-200,0" end="0,0">WindowOpen</animation>
+			<animation effect="slide" time="200" start="0,0" end="-200,0">WindowClose</animation>
+			<control type="image">
+				<left>10</left>
+				<centertop>50%</centertop>
+				<width>30</width>
+				<height>30</height>
+				<texture colordiffuse="$VAR[DialogColor2]">sub-menu-return.png</texture>
+			</control>
+		</control>
+		<control type="group">
+			<visible>!Window.IsVisible(contextmenu) + Control.HasFocus(60)</visible>
+			<animation effect="fade" start="0" end="100" time="200">VisibleChange</animation>
+			<animation effect="slide" time="200" start="-200,0" end="0,0">WindowOpen</animation>
+			<animation effect="slide" time="200" start="0,0" end="-200,0">WindowClose</animation>
+			<control type="image">
+				<left>10</left>
+				<centertop>50%</centertop>
+				<width>30</width>
+				<height>30</height>
+				<texture colordiffuse="$VAR[DialogColor2]">sub-menu-left.png</texture>
 			</control>
+		</control>
 
 		<!-- Options -->
 		<control type="group" id="9002">
@@ -411,7 +447,7 @@
 						<label>19270</label>
 						<visible>Window.IsActive(TVRecordings) | Window.IsActive(RadioRecordings)</visible>
 					</control>
-
+					
 					<control type="image" id="81">
 						<left>0</left>
 						<top>0</top>
@@ -419,6 +455,23 @@
 						<height>10</height>
 						<texture>transparent.png</texture>
 					</control>
+					
+					<!-- Scrollbars -->
+					<control type="radiobutton" id="94">
+						<height>52</height>
+						<label>31005</label>
+						<onclick>Skin.ToggleSetting(Scrollbars)</onclick>
+						<selected>!Skin.HasSetting(Scrollbars)</selected>
+					</control>
+
+
+					<control type="image" id="95">
+						<left>0</left>
+						<top>0</top>
+						<width>410</width>
+						<height>10</height>
+						<texture>transparent.png</texture>
+					</control>
 
 					<!-- Channels -->
 					<control type="button" id="100">
diff --git a/16x9/MyPVRRecordings.xml b/16x9/MyPVRRecordings.xml
index a682d49..fb34991 100755
--- a/16x9/MyPVRRecordings.xml
+++ b/16x9/MyPVRRecordings.xml
@@ -95,7 +95,6 @@
 				<onup>51</onup>
 				<ondown>51</ondown>
 				<onleft>60</onleft>
-				<onright condition="!Skin.HasSetting(KioskMode)">3001</onright>
 				<pagecontrol>60</pagecontrol>
 				<focusposition>3</focusposition>
 				<orientation>vertical</orientation>
@@ -184,7 +183,7 @@
 				<centertop>50%</centertop>
 				<width>20</width>
 				<height>720</height>
-				<onleft>3000</onleft>
+				<onleft condition="!Skin.HasSetting(KioskMode)">3000</onleft>
 				<onright>50</onright>
 				<showonepage>false</showonepage>
 				<orientation>vertical</orientation>
@@ -276,11 +275,28 @@
 					<scrolltime>200</scrolltime>
 					<orientation>vertical</orientation>
 					<itemgap>0</itemgap>
-					<defaultcontrol always="true">99</defaultcontrol>
+					<defaultcontrol always="true">3</defaultcontrol>
 
 					<include content="SideMenuAnimation">
 						<param name="containerID">3000</param>
 					</include>
+					
+					<!-- Scrollbars -->
+					<control type="radiobutton" id="1">
+						<label>31005</label>
+						<height>52</height>
+						<onclick>Skin.ToggleSetting(Scrollbars)</onclick>
+						<selected>!Skin.HasSetting(Scrollbars)</selected>
+					</control>
+
+
+					<control type="image" id="2">
+						<left>0</left>
+						<top>0</top>
+						<width>410</width>
+						<height>10</height>
+						<texture>transparent.png</texture>
+					</control>
 
 					<!-- Sort by -->
 					<control type="button" id="3">
@@ -297,7 +313,7 @@
 						<visible>Control.IsEnabled(4)</visible>
 					</control>
 					<!-- Watched toggle -->
-					<control type="radiobutton" id="10">
+					<control type="button" id="10">
 						<label>20367</label>
 						<height>52</height>
 						<visible>Control.IsEnabled(10)</visible>
diff --git a/16x9/MyPVRSearch.xml b/16x9/MyPVRSearch.xml
index 435b34d..d9186ff 100755
--- a/16x9/MyPVRSearch.xml
+++ b/16x9/MyPVRSearch.xml
@@ -34,7 +34,7 @@
 				<onup>50</onup>
 				<ondown>50</ondown>
 				<onleft>3000</onleft>
-				<onright>70</onright>
+				<onright condition="!Skin.HasSetting(KioskMode)">70</onright>
 				<pagecontrol>70</pagecontrol>
 				<preloaditems>2</preloaditems>
 				<viewtype label="535">list</viewtype>				
@@ -360,11 +360,28 @@
 					<scrolltime>200</scrolltime>
 					<orientation>vertical</orientation>
 					<itemgap>0</itemgap>
-					<defaultcontrol always="true">99</defaultcontrol>
+					<defaultcontrol always="true">3</defaultcontrol>
 
 					<include content="SideMenuAnimation">
 						<param name="containerID">3000</param>
 					</include>
+					
+					<!-- Scrollbars -->
+					<control type="radiobutton" id="1">
+						<label>31005</label>
+						<height>52</height>
+						<onclick>Skin.ToggleSetting(Scrollbars)</onclick>
+						<selected>!Skin.HasSetting(Scrollbars)</selected>
+					</control>
+
+
+					<control type="image" id="2">
+						<left>0</left>
+						<top>0</top>
+						<width>410</width>
+						<height>10</height>
+						<texture>transparent.png</texture>
+					</control>
 
 					<!-- Sort by -->
 					<control type="button" id="3">
diff --git a/16x9/MyPVRTimers.xml b/16x9/MyPVRTimers.xml
index 72eab2b..5985981 100755
--- a/16x9/MyPVRTimers.xml
+++ b/16x9/MyPVRTimers.xml
@@ -22,7 +22,7 @@
 				<onup>50</onup>
 				<ondown>50</ondown>
 				<onleft>3000</onleft>
-				<onright>70</onright>
+				<onright condition="!Skin.HasSetting(KioskMode)">70</onright>
 				<pagecontrol>70</pagecontrol>
 				<preloaditems>2</preloaditems>
 				<viewtype label="535">list</viewtype>				
@@ -275,11 +275,28 @@
 					<scrolltime>200</scrolltime>
 					<orientation>vertical</orientation>
 					<itemgap>0</itemgap>
-					<defaultcontrol always="true">99</defaultcontrol>
+					<defaultcontrol always="true">3</defaultcontrol>
 
 					<include content="SideMenuAnimation">
 						<param name="containerID">3000</param>
 					</include>
+					
+					<!-- Scrollbars -->
+					<control type="radiobutton" id="1">
+						<label>31005</label>
+						<height>52</height>
+						<onclick>Skin.ToggleSetting(Scrollbars)</onclick>
+						<selected>!Skin.HasSetting(Scrollbars)</selected>
+					</control>
+
+
+					<control type="image" id="2">
+						<left>0</left>
+						<top>0</top>
+						<width>410</width>
+						<height>10</height>
+						<texture>transparent.png</texture>
+					</control>
 
 					<!-- Sort by -->
 					<control type="button" id="3">
